Templates/CSharp/Requests/MethodRequestBuilder.cs.tt

```diff
@@ -5,417 +5,119 @@
-            
-            if (isAction)
-            {
-                this.AddParameterToRequestConstructorStringBuilder(requestConstructorParamsStringBuilder, param);
-            }
-            else
-            {
-                this.AddParameterToRequestUrlInitializerStringBuilder(requestUrlInitializerStringBuilder, param, true, first);
-                
-                first = false;
-            }
-        }
-
-        builderConstructorParamsStringBuilder.Remove(builderConstructorParamsStringBuilder.Length - 1, 1);
-        
-        if (isAction)
-        {
-            // We'll only have the trailing comma in the request constructor string builder for actions. Functions add the request parameters to the URL.
-            requestConstructorParamsStringBuilder.Remove(requestConstructorParamsStringBuilder.Length - 1, 1);
-        }
-        else if (hasParameters)
-        {
-            requestUrlInitializerStringBuilder.Append(Environment.NewLine);
-            requestUrlInitializerStringBuilder.Append(Environment.NewLine);
-            requestUrlInitializerStringBuilder.Append("            functionRequestUrl = string.Format(\"{0}({1})\", this.RequestUrl, functionParametersStringBuilder.ToString());");
-        }
-    #>
-
-        public <#=requestBuilderType#>(<#=builderConstructorParamsStringBuilder.ToString()#>)
-            : base(requestUrl, client)
-        {
-            <#=initializersStringBuilder.ToString()#>
-
-        }
-    <#
+#>
+            SetParameter("<#=p.Name#>", <#=p.ParameterName#>, <#=p.IsNullable.ToString().ToLowerInvariant()#>);
+<#
     }
-
-    foreach (var param in method.Parameters)  
-    {
-        var paramTypeString = param.Type.GetTypeString();
-        
-        if (param.IsCollection)
-        {
-            paramTypeString = string.Format("IEnumerable<{0}>", paramTypeString);
-        }
-        else if (!param.Type.IsTypeNullable() && param.IsNullable)
-        {
-            paramTypeString = string.Format("{0}?", paramTypeString);
+#>
         }
-    #>
-
-        /// <summary>
-        /// Gets the value of <#=param.Name.ToCheckedCase()#>.
-        /// </summary>
-        public <#=paramTypeString#> <#=param.Name.ToCheckedCase()#> { get; set; }
-    <#
-    }
-    #>
 
+<#
+}
+#>
         /// <summary>
-        /// Builds the request.
+        /// A method used by the base class to construct a request class instance.
         /// </summary>
+        /// <param name="functionUrl">The request URL to </param>
         /// <param name="options">The query and header options for the request.</param>
-        /// <returns>The built request.</returns>
-        public I<#=requestType#> Request(IEnumerable<Option> options = null)
-        {
-    <#
-        if (!isAction && hasParameters)
+        /// <returns>An instance of a specific request class.</returns>
+        protected override I<#=requestType#> CreateRequest(string functionUrl, IEnumerable<Option> options)
```

Comment:
Ah, nevermind. I didn't realize Request is still there too. Ignore this comment.

---
In reply to: [69044041](https://github.com/microsoftgraph/MSGraph-SDK-Code-Generator/pull/24#discussion_r69044041) [](ancestors = 69044041)